lib/spack/spack/cmd/develop.py

```diff
@@ -123,6 +131,25 @@ def develop(parser, args):
 
     spec = specs[0]
 
+    if args.recursive:
+        concrete_specs = env.all_matching_specs(spec)
+        if not concrete_specs:
+            tty.msg(
+                "No matching specs found in the environment. "
+                "Recursive develop requires a concretized environment"
+            )
+        else:
+            for s in concrete_specs:
+                for parent in s.traverse_edges(direction="parents", root=True):
```

Comment:
use spack.traverse.traverse_nodes(concrete_specs) instead of a double loop and key by dag hash.

Reply:
Do you think there is a better traversal logic I could get to hit the intended intersection of the DAG? I read the warnings about upward traversal, and I do find it concerning.  I could see a visit each node in the DAG and check a condition if there is an accessible and sufficient condition on the specs to say "I depened on spec foo in my subgraph?".


Maybe something like:
``` python
if f"^{dev_spec.name" in parent_spec.fullname":
   develop(parent_spec)
```

Reply:
I think I'll mark this as resolved for now. I don't really know how to test this. I haven't hit an issue with it, but if someone wants to come up with some potential tests then I will be happy to run them.